src/time-slots.ts

```diff
@@ -185,21 +185,31 @@ function _getAvailableTimeSlotsForShift(
 	return timeSlots
 }
 
-/* Filters emcompassed events */
-function _cleanEventList(eventList: DayjsPeriod[]) {
-	return eventList
-		.filter((event) => {
-			for (const otherEvent of eventList) {
-				/* Search for an event with strictly larger boundaries */
-				if (
-					otherEvent.startAt.isBefore(event.startAt)
-					&& otherEvent.endAt.isAfter(event.endAt)
-				) {
-					return false
-				}
-			}
+/* Comparison function to sort DayjsPeriod on start date*/
+function _sortPeriods(periods: DayjsPeriod[]) {
+	return periods.sort((a, b) => (a.startAt.isAfter(b.startAt) ? 1 : -1))
+}
+
+/* Uses a binary search function (O(logN). Event list must be sorted on event.startAt */
+function _findEmcompassingEvent(eventList: DayjsPeriod[], event: DayjsPeriod): boolean {
+	let start = 0
+	let end = eventList.length - 1
+
+	// Iterate while start not meets end
+	while (start <= end) {
+		// Find the mid index
+		const mid = Math.floor((start + end) / 2)
+
+		// If element is present at mid, return True
+		if (eventList[mid].startAt.isBefore(event.startAt)
+			&& eventList[mid].endAt.isAfter(event.endAt)) {
 			return true
-		})
+		// Else look in left or right half accordingly
+		} else if (eventList[mid].startAt < event.startAt) {
+			start = mid + 1
+		} else { end = mid - 1 }
+	}
+	return false
```

Comment:
I think you should extract the binary search in another function, as we have utility and business code mixed here

Reply:
The thing is that the search function is tightly coupled to encompassing event search. Take a loook a the search success condition (ligne 204).
I guess we could imagine a more generic function like _binarySearch(anArrayOfObjects, aValidationFunction, aLeftRightFunction) but I don't really see the point...
Do you have something else in mind ?

Reply:
> I guess we could imagine a more generic function like _binarySearch(anArrayOfObjects, aValidationFunction, aLeftRightFunction) but I don't really see the point...

Yes, that's what I was thinking of. 

The point is to separate the business logic (i.e. filtering the encompassing event) from the utility part (i.e. using a binary search). Because at first sight, it was a bit confusing for me, and it took some time before I understood what's was really going on inside `_findEmcompassingEvent`